examples/PlutusComparison.md

```diff
@@ -0,0 +1,288 @@
+
+We are going to compare two different off-chain contracts 
+both in Plutus and CTL. 
+
+For this discussion, we need to go through an overview of 
+both of them.
+
+
+The purpose of `Contract` in CTL is to bring the capabilities of 
+`QueryMExtended` to the public API. 
+
+The current definition of `Contract` in CTL is :
+
+```PureScript
+type QueryConfig (r :: Row Type) =
+  { ogmiosWs :: OgmiosWebSocket
+  , datumCacheWs :: DatumCacheWebSocket
+  , serverConfig :: ServerConfig
+  , wallet :: Maybe Wallet
+  -- should probably be more tightly coupled with a wallet
+  , usedTxOuts :: UsedTxOuts
+  , networkId :: NetworkId
+  , slotConfig :: SlotConfig
+  , logLevel :: LogLevel
+  | r
+  }
+
+type DefaultQueryConfig = QueryConfig ()
+
+type QueryM (a :: Type) = ReaderT DefaultQueryConfig (LoggerT Aff) a
+
+type QueryMExtended (r :: Row Type) (a :: Type) =
+  ReaderT 
+    (QueryConfig r)
+    (LoggerT Aff)
+    a
+
+newtype Contract (r :: Row Type) (a :: Type) = Contract (QueryMExtended r a)
+```
+
+
+In CTL we have a general environment type `forall r . QueryConfig r`, it stores
+the needed parameters to connect to an `Ogmios` server  and
+uses the PureScript native row polymorphism to make it extensible for both 
+CTL developers and users.
+
+The parameter `a` as in `Plutus` refers to a return value wrapped by `Contract`.
+
+
+Note that in Plutus right now we have the following definition for `Contract`:
+
+```Haskell
+type ContractEffs w e =
+    '[ Error e
+    ,  LogMsg Value
+    ,  Writer w
+    ,  Checkpoint
+    ,  Resumable PABResp PABReq
+    ]
+
+type ContractEnv = (IterationID, RequestID)
+
+newtype Contract w (s :: Row *) e a = Contract { unContract :: Eff (ContractEffs w e) a }
+  deriving newtype (Functor, Applicative, Monad)
+```
+
+The Plutus `Contract` environment is specialized to just two values and is fixed. 
+Also, Plutus `Contract` uses a phantom type to allow users 
```

Comment:
The phantom type parameter `s` is for the contract schema, which CTL doesn't have